src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -84,12 +84,53 @@ string NormalizedRelativePath(const string& cwd, const string& path) {
       continue;
     }
     if (*iter == "..") {
-      iter = segments.erase(iter - 1, iter + 1);
+      // If the previous segment has any one of the following characters,
+      // don't erase the whole previous segment but instead erase only the
+      // relevant portions of the previous segment.
+      auto prev_iter = iter - 1;
+      const size_t single_quote_pos = prev_iter->find_last_of("'");
+      const size_t space_pos = prev_iter->find_last_of(" ");
+      const size_t double_quote_pos = prev_iter->find_last_of("\"");
+      // The precedence of checks below should be maintained as
+      // 1. double_quote_pos
+      // 2. single_quote_pos
+      // 3. space_pos
+      // This is because of the way Android Platform 7 constructs build commands.
+      if (double_quote_pos != string::npos) {
```

Comment:
Can we get the max index of where any of the delimiters exist and use that to split?

Reply:
Done